plugins/scaffolder-backend/src/scaffolder/actions/builtin/publish/githubPullRequest.ts

```diff
@@ -255,15 +255,17 @@ export const createPublishGithubPullRequestAction = ({
           {
             // See the properties of tree items
             // in https://docs.github.com/en/rest/reference/git#trees
-            mode: file.executable ? '100755' : '100644',
+            mode: file.symlink ? '120000' : (file.executable ? '100755' : '100644'),
```

Comment:
It's not much better but `file.symlink` could perhaps be replaced with `file.mode` to cut down on this whole "ternary within a ternary" business. You'd still end up with a weird check below though like `file.mode === 120000 ? 'utf-8' : 'base64` which... I dunno, maybe expresses better what Git is up to

Reply:
This is probably fine, just move the logic into a little helper function